zerver/views/streams.py

```diff
@@ -318,13 +326,39 @@ def add_subscriptions_backend(
     if len(principals) > 0:
         if user_profile.realm.is_zephyr_mirror_realm and not all(stream.invite_only for stream in streams):
             return json_error(_("You can only invite other Zephyr mirroring users to invite-only streams."))
-        subscribers = set(principal_to_user_profile(user_profile, principal) for principal in principals)
+        subscribers = set(user_email_to_user_profile(user_profile, principal) for principal in principals)
     else:
         subscribers = set([user_profile])
 
+    for stream in created_streams:
+        # You can not create private stream without stream admin.
+        if stream.invite_only and len(stream_admin_ids) == 0:
+                return json_error(_("Private stream must require at least one admin."))
+
+    if len(stream_admin_ids) > 0:
+        stream_admins = set(
+            [user_id_to_user_profile(admin_id) for admin_id in stream_admin_ids])
+    else:
+        stream_admins = set()
+
+    # Check if stream admin subscribed to stream or not.
+    stream_ids = set([stream.id for stream in streams])
+    for stream_admin in stream_admins:
+        # If stream admin is not already subscribed to stream and not included
+        # in subscribers list, raise an error.
+        admin_subscribed_streams = do_get_streams(stream_admin, include_default=True)
+        admin_subscribed_stream_ids = set([admin_subscribed_stream['stream_id']
+                                          for admin_subscribed_stream in admin_subscribed_streams])
+        if stream_admin not in subscribers and not admin_subscribed_stream_ids.issubset(stream_ids):
+            return json_error(_("Stream admin '%s' must subscribe to stream.") % (stream_admin.email))
+
```

Comment:
Here, we want to check that whether stream admin is subscribed to stream or not.
* Either stream admin should be already subscribed to stream.
* Or stream admin should be in `subscribers` list(who will be subscribed to stream)